src/pages/academy/Academy.tsx

```diff
@@ -37,6 +37,7 @@ const CourseSelectingAcademy: React.FC = () => {
   const { courseId } = useSession();
   const { courseId: routeCourseIdStr } = useParams<{ courseId?: string }>();
   const routeCourseId = routeCourseIdStr != null ? parseInt(routeCourseIdStr, 10) : undefined;
+  const enableExamMode = useTypedSelector(state => state.session.enableExamMode);
 
```

Comment:
`useSession` is a wrapper around `useTypedSelector(state => state.session)` among other things

```suggestion
  const { courseId, enableExamMode } = useSession();
  const { courseId: routeCourseIdStr } = useParams<{ courseId?: string }>();
  const routeCourseId = routeCourseIdStr != null ? parseInt(routeCourseIdStr, 10) : undefined;

```